py/desispec/util.py

```diff
@@ -655,6 +655,42 @@ def parse_fibers(fiber_string, include_end=False) :
     """
     return parse_int_args(fiber_string, include_end)
 
+def parse_nights(nights_string, include_end=False) :
+    """
+    Short func that parses a string containing a comma separated list of
+    YYYYMMDD, which can include ":" or ".." or "-" labeled ranges
+
+    Args:
+        nights_string (str) : list of integers or integer ranges
+
+    Options:
+        include_end (bool): if True, include end-value in ranges
+
+    Returns (array 1-D):
+        1D numpy array listing all of the integers given in the list,
+        including enumerations of ranges given.
+
+    Note: this follows python-style ranges, i,e, 1:5 or 1..5 returns 1, 2, 3, 4
+    unless `include_end` is True, which then returns 1,2,3,4,5
```

Comment:
At first I was going to object that this duplicates `parse_int_args`, but then I saw that you are using `parse_int_args` and then dropping entries that aren't valid dates, which is nice to handling ranges that span month boundaries.  Please update docstring to mention that, since these examples actually don't apply to this function (1:5 would drop all entries).

For the record: I checked and this handles leap-years correctly :)

Reply:
done